label_studio/jwt_auth/views.py

```diff
@@ -0,0 +1,157 @@
+        },
+    )
+    def post(self, request, *args, **kwargs):
+        return super().post(request, *args, **kwargs)
+
+
+@method_decorator(
+    name='get',
+    decorator=swagger_auto_schema(
+        tags=['JWT'],
+        operation_summary='List API tokens',
+        operation_description='List all API tokens for the current user.',
+    ),
+)
+@method_decorator(
+    name='post',
+    decorator=swagger_auto_schema(
+        tags=['JWT'],
+        operation_summary='Create API token',
+        operation_description='Create a new API token for the current user.',
+    ),
+)
+class LSAPITokenView(generics.ListCreateAPIView):
+    permission_classes = [IsAuthenticated]
+    token_class = LSAPIToken
+
+    def get_queryset(self):
+        """Returns all non-expired non-blacklisted tokens for the current user.
+
+        The `list` method handles filtering for refresh tokens (as opposed to access tokens),
+        since simple-jwt makes it hard to do this at the DB level."""
+        current_blacklisted_tokens = BlacklistedToken.objects.filter(token__expires_at__gt=datetime.now()).values_list(
+            'token_id', flat=True
+        )
+        return OutstandingToken.objects.filter(user_id=self.request.user.id, expires_at__gt=datetime.now()).exclude(
+            id__in=current_blacklisted_tokens
+        )
+
+    def list(self, request, *args, **kwargs):
+        all_tokens = self.get_queryset()
+
+        def _maybe_get_token(token: OutstandingToken):
+            try:
+                return TruncatedLSAPIToken(str(token.token))
+            except (TokenError, TokenBackendError) as e:  # expired/invalid token
+                logger.info('JWT API token validation failed: %s', e)
+                return None
+
+        # Annoyingly, token_type not stored directly so we have to filter it here.
+        # Shouldn't be many unexpired tokens to iterate through.
+        token_objects = list(filter(None, [_maybe_get_token(token) for token in all_tokens]))
+        refresh_tokens = [tok for tok in token_objects if tok['token_type'] == 'refresh']
+
+        serializer = self.get_serializer(refresh_tokens, many=True)
+        data = serializer.data
+        return Response(data)
+
+    def get_serializer_class(self):
+        if self.request.method == 'POST':
+            return LSAPITokenCreateSerializer
+        return LSAPITokenListSerializer
+
+    def perform_create(self, serializer):
+        token = self.token_class.for_user(self.request.user)
+        serializer.instance = token
+
+
+class LSTokenBlacklistView(TokenViewBase):
+    _serializer_class = 'jwt_auth.serializers.LSAPITokenBlacklistSerializer'
+
+    @swagger_auto_schema(
+        tags=['JWT'],
+        operation_summary='Blacklist a JWT refresh token',
+        operation_description='Adds a JWT refresh token to the blacklist, preventing it from being used to obtain new access tokens.',
+        responses={
+            204: 'Token was successfully blacklisted',
```

Comment:
I figured a 204 is appropriate here, since this is more-or-less a "delete", but lmk if there's a more appropriate response here